python/langsmith/evaluation/_arunner.py

```diff
@@ -642,6 +651,56 @@ async def astart(self) -> _AsyncExperimentManager:
             upload_results=self._upload_results,
         )
 
+    async def awith_predictions_and_evaluators(
+        self,
+        target: ATARGET_T,
+        evaluators: Sequence[Union[EVALUATOR_T, AEVALUATOR_T]],
+        /,
+        max_concurrency: Optional[int] = None,
+    ) -> _AsyncExperimentManager:
+        """Run predictions and evaluations in a single pipeline.
+        
+        This allows evaluators to process results as soon as they're available from
+        the target function, rather than waiting for all predictions to complete first.
+        """
+        evaluators = _resolve_evaluators(evaluators)
+        
+        if not hasattr(self, '_evaluator_executor'):
+            self._evaluator_executor = cf.ThreadPoolExecutor(max_workers=4)
```

Comment:
I don't like doing this, but I couldn't figure out another way to pass an executor that doesn't trigger a `raise RuntimeError('cannot schedule new futures after shutdown')` from `_log_evaluation_feedback`. I also don't think I ever delete this executor, but I am not 100% sure where to do so